### PR TITLE
Updating the RHESSI colortable 

### PR DESCRIPTION
**Pull request by hayesla**
This PR is an update to the colormap `rhessi` from using `plt.cm.jet` to using the mission colortable following #4606 

There are two colortables added here - the mission defined one and also the IDL colortable `STD GAMMA II` . Both are added as both are used by the community. 

The colortable labelled `rhessi` is the mission colormap and the colortable labeled `rhessi2` is the `STD GAMMA II` - this is probably not the best naming convention - maybe `rhessi2` can just be called `std_gamma` but i think having rhessi labelled within its name would be good. 

The default colormap for RHESSIMap is now the mission colortable.

Example:

```python
>>> import sunpy.data.sample
>>> import sunpy.map
>>> import matplotlib.pyplot as plt 

>>> rhessi_map = sunpy.map.Map(sunpy.data.sample.RHESSI_IMAGE)
>>> rhessi_map.peek()
```
Figure_1

or you can use the other `rhessi2` via:
```python
>>> rhessi_map.peek(cmap="rhessi2")
```
Figure_2

### CODE REVIEW

**Comment by nabobalis** [comment]
I do prefer the name "std_gamma" over "rhessi2". Could it be "std_gamma_2" instead? 

Have to see what others prefer.

**Comment by kakirastern** [comment]
Have a feeling you should try out the free version of PyCharm to minimise the risks of code style mishaps...

File: sunpy/visualization/colormaps/cm.py [hunk 1]
```diff
@@ -128,7 +130,9 @@
     'irissjiFUV': ct.iris_sji_color_table('FUV'),
     'irissjiNUV': ct.iris_sji_color_table('NUV'),
     'irissjiSJI_NUV': ct.iris_sji_color_table('SJI_NUV'),
-    'kcor': kcor
+    'kcor': kcor,
+    'rhessi': rhessi,
+    'rhessi2': rhessi2
```
Comment (nabobalis) [inline sunpy/visualization/colormaps/cm.py:135]:
```suggestion
    'rhessi2': rhessi2,
```

File: sunpy/visualization/colormaps/color_tables.py [hunk 2]
```diff
@@ -13,7 +13,8 @@
 __all__ = [
     'aia_color_table', 'sswidl_lasco_color_table', 'eit_color_table',
     'sxt_color_table', 'xrt_color_table', 'trace_color_table',
-    'sot_color_table', 'hmi_mag_color_table', 'suvi_color_table'
+    'sot_color_table', 'hmi_mag_color_table', 'suvi_color_table',
+    'rhessi_color_table', 'rhessi_color_table2'
```
Comment (nabobalis) [inline sunpy/visualization/colormaps/color_tables.py:17]:
```suggestion
    'rhessi_color_table', 'rhessi_color_table2',
```

**Comment by nabobalis** [comment]
Oh the figure test for the color tables will need updating. Not sure about the other tests.

**Comment by hayesla** [comment]
> Oh the figure test for the color tables will need updating. Not sure about the other tests.
 
Thanks @nabobalis  - Can you point me to where the figure tests are?

**Comment by nabobalis** [comment]
> > Oh the figure test for the color tables will need updating. Not sure about the other tests.
> 
> Thanks @nabobalis - Can you point me to where the figure tests are?

You will want to get the hash files from:
https://app.circleci.com/pipelines/github/sunpy/sunpy/8729/workflows/a7028b6d-49ba-48a6-a75e-b7514f2164c6/jobs/59906/artifacts
and
https://app.circleci.com/pipelines/github/sunpy/sunpy/8729/workflows/a7028b6d-49ba-48a6-a75e-b7514f2164c6/jobs/59905/artifacts
and update them in your PR.

File: sunpy/visualization/colormaps/color_tables.py [hunk 3]
```diff
@@ -299,6 +300,12 @@ def suvi_color_table(wavelength: u.angstrom):
     return _cmap_from_rgb(r, g, b, 'GOES-R SUVI {:s}'.format(str(wavelength)))
 
 
+def rhessi_color_table():
+    return cmap_from_rgb_file(f"rhessi", f"rhessi.csv")
```
Comment (nabobalis) [inline sunpy/visualization/colormaps/color_tables.py:304]:
```suggestion
    return cmap_from_rgb_file(f"rhessi", "rhessi.csv")
```

File: sunpy/visualization/colormaps/color_tables.py [hunk 4]
```diff
@@ -299,6 +300,12 @@ def suvi_color_table(wavelength: u.angstrom):
...
+
+def rhessi_color_table2():
+    return cmap_from_rgb_file(f"rhessi2", f"rhessi2.csv")
```
Comment (nabobalis) [inline sunpy/visualization/colormaps/color_tables.py:307]:
```suggestion
    return cmap_from_rgb_file("rhessi2", "rhessi2.csv")
```

File: docs/guide/data_types/maps.rst [hunk 5]
```diff
@@ -388,11 +388,12 @@ long as it recognizes the instrument. To see what colormaps are available::
...
     'sdoaia4500', 'sohoeit171', 'sohoeit195', 'sohoeit284', 'sohoeit304', 'soholasco2',
     'soholasco3', 'sswidlsoholasco2', 'sswidlsoholasco3', 'stereocor1',
-    'stereocor2', 'stereohi1', 'stereohi2', 'rhessi', 'yohkohsxtal',
+    'stereocor2', 'stereohi1', 'stereohi2', 'yohkohsxtal',
     'yohkohsxtwh', 'hinodexrt', 'hinodesotintensity', 'trace171', 'trace195',
     'trace284', 'trace1216', 'trace1550', 'trace1600', 'trace1700', 'traceWL',
     'hmimag', 'irissji1330', 'irissji1400', 'irissji1600', 'irissji2796',
-    'irissji2832', 'irissji5000', 'irissjiFUV', 'irissjiNUV', 'irissjiSJI_NUV', 'kcor'])
+    'irissji2832', 'irissji5000', 'irissjiFUV', 'irissjiNUV', 'irissjiSJI_NUV', 'kcor',
+    'rhessi', 'rhessi2'])
```
Comment (nabobalis) [inline docs/guide/data_types/maps.rst:396]: Just needs to be updated again for the colormap rename.

Comment (hayesla) [reply docs/guide/data_types/maps.rst]: yep sorry - friday night brain mush

**Comment by nabobalis** [comment]
Just a changelog and this is good to go for me.

File: changelog/4665.feature.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Add two new colormaps that are used for plotting RHESSI maps.
```
Comment (nabobalis) [inline changelog/4665.feature.rst:1]:
```suggestion
Add two new colormaps (``rhessi`` and ``std_gamma_2``) that are used for plotting RHESSI maps.
```

**Review by nabobalis — APPROVED** [review]

**Comment by kakirastern** [comment]
Looks great 👍🏼

**Review by wtbarnes — APPROVED** [review]

**Comment by Laubeee** [comment]
@hayesla works at my end! Thanks!

**Pull request merged**